inc/planningcsv.class.php

```diff
@@ -0,0 +1,191 @@
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ * Planning CSV export Class
+**/
+class PlanningCsv extends CommonGLPI {
+
+   private $users_id;
+   private $groups_id;
+   private $limititemtype;
+
+   private $eol = "\r\n";
+   private $quote = '"';
+   private $titles = [];
+   private $lines = null;
+   private $filename = 'planning.csv';
+
+   /**
+    * @param integer $who          user ID
+    * @param integer $who_group    group ID
+    * @param string $limititemtype itemtype only display this itemtype (default '')
+   **/
+
+   public function __construct($who, $who_group = null, $limititemtype = '') {
+      $this->users_id = $who;
+      $this->groups_id = $who_group;
+      $this->limititemtype = $limititemtype;
+      $this->titles = [
+         __('Actor'),
+         __('Title'),
+         __('Item type'),
+         __('Item id'),
+         __('Begin date'),
+         __('End date')
+      ];
+   }
+
+   /**
+    * Build CSV lines
+    *
+    * @return void
+    */
+   private function buildLines() {
+      global $CFG_GLPI;
+
+      $interv = [];
+      $this->lines = [];
+      $begin  = time()-MONTH_TIMESTAMP*12;
+      $end    = time()+MONTH_TIMESTAMP*12;
+      $begin  = date("Y-m-d H:i:s", $begin);
+      $end    = date("Y-m-d H:i:s", $end);
+      $params = [
+         'who'       => $this->users_id,
+         'who_group' => $this->groups_id,
+         'whogroup'  => $this->groups_id,
+         'begin'     => $begin,
+         'end'       => $end
+      ];
+
+      if (empty($this->limititemtype)) {
+         foreach ($CFG_GLPI['planning_types'] as $itemtype) {
+            $interv = array_merge($interv, $itemtype::populatePlanning($params));
+         }
+      } else {
+         $interv = $this->limititemtype::populatePlanning($params);
+      }
+
+      if (count($interv) > 0) {
+         foreach ($interv as $val) {
+            $dateBegin = new DateTime($val["begin"]);
+            $dateBegin->setTimeZone(new DateTimeZone('UTC'));
+
+            $dateEnd = new DateTime($val["end"]);
+            $dateEnd->setTimeZone(new DateTimeZone('UTC'));
+
+            $itemtype = new $val['itemtype'];
+
+            $user = new User();
+            $user->getFromDB($val['users_id']);
```

Comment:
What do we want to be exported for `actor` field ? `users_id` field value correspond to the author of the item, not the assigned user.

- For `CommonITILTask`, assigned user is in `users_id_tech` field,
- For `ProjectTask` and `Reminder`, target users are not available in items returned by `populatePlanning`.

Reply:
ping @orthagh

Reply:
No i think it's ok, in the current case, users_id came from populatePlanning (in $val var).